formats/cbor/commonMain/src/kotlinx/serialization/cbor/internal/Streams.kt

```diff
@@ -33,6 +33,10 @@ internal class ByteArrayInput(private var array: ByteArray) {
         return copied
     }
 
+    fun skip(length: Int) {
+        require(length >= 0) { "Cannot skip $length bytes" }
```

Comment:
Either length cannot be negative here or `SerializationException` should be thrown as well

Reply:
Good call, I've dropped the negative length check as the reading of the length (in `elementLength()` function) only reads headers that would **not** result in `readNumber` returning a negative number.